src/Plugin/FieldEncryptionProvider/EncryptProviderBase.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\field_encrypt\Plugin\FieldEncryptionProvider\EncryptProviderBase.
+ */
+
+namespace Drupal\field_encrypt\Plugin\FieldEncryptionProvider;
+
+use Drupal\Core\Form\FormStateInterface;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\field_encrypt\FieldEncryptionProviderBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\encrypt\EncryptionProfileManagerInterface;
+use Drupal\encrypt\EncryptServiceInterface;
+
+/**
+ * A base FieldEncryptionProvider class that integrates with Encrypt service.
```

Comment:
As per my previous comments, I think this may be overkill. 

When would some Encryption Profile _not_ work? 

I still think, at it's core, field encrypt performs encrypt and decrypt operations. Encrypt API pretty much enforces that...